hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/HoodieInternalRowUtils.scala

```diff
@@ -219,11 +237,34 @@ object HoodieInternalRowUtils {
     schemaMap.get(schema)
   }
 
+  def existField(structType: StructType, name: String): Boolean = {
+    val stack = new mutable.Stack[(StructType, String)]
```

Comment:
Instead of stacking, can we just split the name once and retrieve nested one in a single pass?

Reply:
Oh, i totally forgot that we actually have `composeNestedFieldPath` that has already been optimized for such retrievals. Let's modify it to be able to use it here and we should be good to go